packages/components/src/chart/d3chart/utils.js

```diff
@@ -299,6 +299,21 @@ export const calculateXTicksIncrementFactor = ( uniqueDates, maxTicks ) => {
 	return factors.find( f => uniqueDates.length / f < maxTicks );
 };
 
+/**
+ * Given an array of dates, returns true if the first and last one belong to the same day.
+ * @param {array} dates - an array of dates
+ * @returns {boolean} whether the first and last date are different hours from the same date.
+ */
+const areDatesInTheSameDay = dates => {
```

Comment:
Since this method is working off of UTC - how will this work properly with the site's timezone?

Reply:
Good catch. That was a bug, I was manually modifying the date to convert it to UTC and then getting its UTC values. Of course, that shouldn't be needed.

I fixed it here: 000151b488de6e6e64c71b20d1b1c8931bb562aa. It should work exactly the same but code is simpler.